torchci/pages/api/drci/drci.ts

```diff
@@ -78,6 +79,26 @@ export default async function handler(
       prNumber ? [parseInt(prNumber as string)] : []
     );
     res.status(200).json(failures);
+  } else if (authorization) {
+    // Check user rate limit
```

Comment:
I think this block could be rewrite to avoid code duplication.  Something likes this:

```
const authorization = req.headers.authorization;
// Perform authorization check + rate limit check for user + increase rate limit
if (await drCINotAuthorizeOrRateLimitExceeded(authorization)) {
  return res.status(429).end();
}

const { prNumber } = req.query;
const { repo }: UpdateCommentBody = JSON.parse(req.body);

const octokit = await getOctokit(OWNER, repo);
const failures = await updateDrciComments(
  octokit,
  repo,
  prNumber ? [parseInt(prNumber as string)] : []
);
// Have a consistent API is good, just return the failures even if the client doesn't care
res.status(200).json(failures);
```

Reply:
Reduced duplication but in a different way